nixos/modules/services/misc/matrix-registration.nix

```diff
@@ -0,0 +1,218 @@
+            description = "Where to store the database";
+          };
+          host = mkOption {
+            type = types.str;
+            default = "localhost";
+            description = "What host to listen on";
+          };
+          port = mkOption {
+            type = types.port;
+            default = 5000;
+            description = "What port to listen on";
+          };
+          rate_limit = mkOption {
+            type = types.listOf types.str;
+            default = [ "100 per day" "10 per minute" ];
+            description =
+              "How often is one IP allowed to access matrix-registration";
+          };
+          allow_cors = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Allow Cross Origin Resource Sharing";
+          };
+          ip_logging = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Save IPs in the database";
+          };
+          logging = mkOption {
+            type = types.attrs;
+            description = "Python logging configuration";
+            default = {
+              disable_existing_loggers = false;
+              version = 1;
+              root = {
+                level = "DEBUG";
+                handlers = [ "console" "file" ];
+              };
+              formatters = {
+                brief = { format = "%(name)s - %(levelname)s - %(message)s"; };
+                precise = {
+                  format =
+                    "%(asctime)s - %(name)s - %(levelname)s - %(message)s";
+                };
+              };
+              handlers = {
+                console = {
+                  class = "logging.StreamHandler";
+                  level = "INFO";
+                  formatter = "brief";
+                  stream = "ext://sys.stdout";
+                };
+                file = {
+                  class = "logging.handlers.RotatingFileHandler";
+                  formatter = "precise";
+                  level = "INFO";
+                  filename = "${dataDir}/matrix-registration.log";
+                  maxBytes = 10485760;
+                  backupCount = 3;
+                  encoding = "utf8";
+                };
+              };
+            };
+          };
+          password.min_length = mkOption {
+            type = types.ints.positive;
+            default = 8;
+            description = "Minimum password length for the registered user";
+          };
+        };
+      };
+    };
+
+    credentialsFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        File containing variables to be passed to the matrix-registration service,
+        in which secret tokens can be specified securely by defining values for
+        <literal>REGISTRATION_SHARED_SECRET</literal>,
+        <literal>ADMIN_SECRET</literal>
+      '';
+    };
+
+    serviceDependencies = mkOption {
```

Comment:
I don't think we should have this option. We should infer as much as we can from existing configuration and set reasonable defaults. If the user has advanced needs they can set `after` and `wants` themselves. At what point do we draw the line in creating options that merely duplicate other options, and are only useful when the user is doing something fairly custom.

That is my opinion at least 🤷‍♂️

Reply:
Right that makes sense to me